code/modules/arcade/mob_hunt/mob_datums.dm

```diff
@@ -0,0 +1,549 @@
+	var/base_health = 5						//base health of the mob for battling (effectively max health at level 0)
+	var/attack_multiplier = 1				//how much additional damage per level the mob deals (level * attack_multiplier)
+	var/health_multiplier = 1				//how much additional health per level the mob gets (level * health_multiplier) for calculating max health
+	var/health = list(0, 0)					//health[1] is the mob's current health, health[2] is the mob's max health (calculated in New())
+
+	//SPAWN PREFERENCES AND VARIABLES
+	var/list/area_blacklist = list()		//list of areas this mob can NOT spawn in (such as the bridge)
+	var/list/turf_blacklist = list()		//list of turfs this mob can NOT spawn on (such as wood floors)
+	var/list/area_whitelist = list()		//list of areas this mob is more likely to spawn in (can be used to reinclude subtypes of blacklisted areas)
+	var/list/turf_whitelist = list()		//list of turfs this mob is more likely to spawn on (can be used to reinclude subtypes of blacklisted turfs)
+	var/turf/spawn_point					//gets set and sent to the game server to spawn its avatar (generated in select_spawn or assigned via set_trap)
+	var/lifetime = 6000						//number of deciseconds the mob will remain before despawning (REMEMBER: DECISECONDS! So 6000 is 600 seconds which is 10 minutes)
+	var/is_shiny = 0						//if this gets set at spawn (super rare), the mob is considered "shiny" and will use the shiny icon_state and holographic cards
+
+	//the icon file for mob_hunt stuff is 'icons/effects/mob_hunt.dmi' so reference that for the following vars
+	var/icon_state_normal = "placeholder"	//the icon_state for this mob's normal version
+	var/icon_state_shiny = "placeholder"	//the icon_state for this mob's rare shiney version
+
+	var/is_trap = 0							//if this gets set, the mob is a booby-trap and will electrocute any players that dare attempt to catch it
+
+/datum/mob_hunt/New(set_trap = 0, turf/trap_turf = null, no_register = 0)
+	if(set_trap)
+		level = max_level
+		is_trap = 1
+		spawn_point = trap_turf
+	else
+		level = rand(min_level, max_level)
+	if(prob(1) && prob(1))
+		is_shiny = 1
+	health[2] = base_health + (level * health_multiplier)
+	if(no_register)		//for booster pack cards
+		return
+	if(mob_hunt_server)
+		if(set_trap)
+			if(mob_hunt_server.register_trap(src))
+				return
+		else if(select_spawn())
+			if(mob_hunt_server.register_spawn(src))
+				return
+	qdel(src)	//if you reach this, the datum is just pure clutter, so delete it
+
+/datum/mob_hunt/proc/select_spawn()
+	var/list/possible_areas = list()
+	for(var/areapath in the_station_areas)
+		if(istype(areapath, /area/holodeck))	//don't allow holodeck areas as possible spawns since it will allow it to spawn in the holodeck rooms on z2 as well
+			continue
+		possible_areas += typesof(areapath)
+	for(var/areapath in area_blacklist)
+		possible_areas -= typesof(areapath)
+	for(var/areapath in area_whitelist)
```

Comment:
Unless I'm missing something, this means the increase in spawn chance will be incredibly tiny if whitelisted areas are just listed twice.